### PR TITLE
Works tabs toggle

### LINKED ISSUE #8909: Implement the works 'tabbed' navigation designs
Designs:
- https://www.sketch.com/s/1cc6a024-668b-4d76-9617-115910d3a797
- https://www.sketch.com/s/e870d280-781b-48f8-942b-edfd985d4235

- This should be developed behind a toggle that can later be used for an a/b test

Assumptions:
- all tabs are only visually tabs, they are actually navigation with links to new urls

### PR DESCRIPTION
**Pull request by gestchild**
Relates to #8910 and fixes #8909

- Adds tabbed navigation to the works page and item page behind a toggle.
- This will allow us to explore if/how this should work - e.g. what should happen in the case of audio/video.
- In combination with #8958, we'll be able to check @taceybadgerbrook can track what she needs and will be able to change the toggle to function as an A/B test when required

Work page:
Screenshot 2022-12-05 at 16 17 55

Item page:
Screenshot 2022-12-05 at 16 16 54

### CODE REVIEW

File: catalogue/webapp/components/WorkTabbedNav/WorkTabbedNav.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { Work } from '@weco/common/model/catalogue';
+import { FunctionComponent } from 'react';
+import SubNavigation from '@weco/common/views/components/SubNavigation/SubNavigation';
+import { listView, eye } from '@weco/common/icons';
+import { toLink as itemLink } from '@weco/common/views/components/ItemLink/ItemLink';
+
+type Props = {
+  work: Work;
+  selected: string;
```
Comment (alexwlchan) [inline catalogue/webapp/components/WorkTabbedNav/WorkTabbedNav.tsx:9]: Could we use a more specific type here e.g.

```suggestion
  selected:'catalogueDetails' | imageViewer';
```

to help us when we want to create instances of this?

**Review by alexwlchan — APPROVED** [review]
A couple of questions and suggestions, but mostly looks good. 💪

File: catalogue/webapp/components/Work/Work.tsx [hunk 2]
```diff
@@ -34,11 +36,11 @@ const WorkDetailsWrapper = styled(Space).attrs({
   flex: 1;
 `;
 
-const Container = styled.div.attrs({
+export const Container = styled.div.attrs({
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply catalogue/webapp/components/Work/Work.tsx:39]: I'm thinking if the only thing making this styled component is the class, I'd just use 
`<div className="container">` or yeah maybe it becomes a common Component with the styles from `.container` moved into it, but that would be changing it across the platform.

Same for Grid below. 

Might be worth deciding on a strategy for this kind of thing in our next chat?

File: catalogue/webapp/components/WorkTabbedNav/WorkTabbedNav.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            },
+          },
+          name: 'Catalogue details',
+          icon: listView,
+        },
+        {
+          id: 'imageViewer',
+          url: itemUrl,
+          name: 'Image viewer',
+          icon: eye,
```
[the remark this answers is not in the document]
Comment (gestchild) [reply catalogue/webapp/components/WorkTabbedNav/WorkTabbedNav.tsx:37]: yeah we might or we might not have the tab for audio/visual stuff. The aim is to get this first iteration out behind a toggle to help with discussing issues like this.

File: common/views/components/SubNavigation/SubNavigation.tsx [hunk 4]
```diff
@@ -36,7 +47,12 @@ const SubNavigation: FunctionComponent<Props> = ({
           const isSelected = currentSection === item.id;
           return (
             <Tab key={item.id}>
-              <Link scroll={false} passHref href={item.url}>
+              <Link
+                scroll={false}
+                passHref
+                href={typeof item.url === 'string' ? item.url : item.url.href}
+                as={typeof item.url === 'string' ? undefined : item.url.as}
```
[the remark this answers is not in the document]
Comment (gestchild) [reply common/views/components/SubNavigation/SubNavigation.tsx:54]: Probably not hard, we may end up throwing this away though, so I might leave as is for now.

File: catalogue/webapp/components/WorkTabbedNav/WorkTabbedNav.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexwlchan) [reply catalogue/webapp/components/WorkTabbedNav/WorkTabbedNav.tsx:37]: Sounds good, let's leave it as-is.

**Review by rcantin-w — APPROVED** [review]
Looks great with the icons, I'm glad it was reusable. Let's keep chatting if anything comes up that would benefit changes on the Search side as well!

**Pull request merged**